tests/integrations/cookiecutter/test_XMLExtension.py

```diff
@@ -24,3 +24,43 @@ def test_bool_attr(value, expected):
     env.filters = {}
     XMLExtension(env)
     assert env.filters["bool_attr"](value) == expected
+
+
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        # No special characters
+        ("Hello World", "Hello World"),
+        # Ampersands escaped
+        ("Hello & World", "Hello &amp; World"),
+        # Less than
+        ("Hello < World", "Hello &lt; World"),
+        # Greater than
+        ("Hello > World", "Hello &gt; World"),
+    ],
+)
+def test_xml_escape(value, expected):
+    env = MagicMock()
+    env.filters = {}
+    XMLExtension(env)
+    assert env.filters["xml_escape"](value) == expected
+
+
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        # A single quote wrapped in double quotes
+        ("Hello ' World", '"Hello \' World"'),
+        # A double quote wrapped in single quotes
+        ('Hello " World', "'Hello \" World'"),
+        # A double quote and a single quote wrapped in single quotes
+        ("Hello \" And ' World", '"Hello &quot; And \' World"'),
+        # A double quote and a single quote wrapped in double quotes
+        ("Hello \" Or ' World", '"Hello &quot; Or \' World"'),
```

Comment:
So... Black (our code formatter) is going to fight you here :-) Black will normalise the content of strings to be double quoted if there's any path do that unambiguously. As a result, these last two tests end up being identical.

The good news is that I don't think you actually need to test both. We need *a* legal string that contains both a `"` and a `'` - but as long as we've got that, we don't care how it was escaped or quoted for input. However the value has been passed in (be it TOML or Python), it will need to be correctly escaped to pass the parser that read it - so additional checks beyond that are more a test of TOML and Python parsers than they are of your new XML escaping filter.
```suggestion
        # A double quote and a single quote in the same string value
        ("Hello \" And ' World", '"Hello &quot; And \' World"'),
```